### ISSUE TITLE
t/11-replay-lwp.t is using system perl

### ISSUE THREAD

**Issue by eserte** [hoelzro/plack-middleware-recorder#9]
The t/11-replay-lwp.t test fails on most systems, see http://matrix.cpantesters.org/?dist=Plack-Middleware-Recorder%200.04
```
Can't locate HTTP/Message/PSGI.pm in @INC (you may need to install the HTTP::Message::PSGI module) (@INC contains: /home/cpansand/.cpan/build/Plack-Middleware-Recorder-0.04-lJt5iU/blib/lib /home/cpansand/.cpan/build/Plack-Middleware-Recorder-0.04-lJt5iU/blib/arch /usr/local/lib/perl5/site_perl/mach/5.18 /usr/local/lib/perl5/site_perl /usr/local/lib/perl5/5.18/mach /usr/local/lib/perl5/5.18 /usr/local/lib/perl5/site_perl/5.18 /usr/local/lib/perl5/site_perl/5.18/mach .) at bin/plack-replay line 7.
BEGIN failed--compilation aborted at bin/plack-replay line 7.

#   Failed test 'Using LWP::UserAgent to access an external service should succeed'
#   at t/11-replay-lwp.t line 23.
#                   ''
#     doesn't match '(?^:200 OK)'
# Looks like you failed 1 test of 1.
t/11-replay-lwp.t ................. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
```
The `@INC` contents look like the system perl is used, not the perl which started the build. Probably you have to use `$^X` at some place.

**Comment by hoelzro** [comment]
Hi @eserte, thanks for bringing this to my attention!  I can't seem to reproduce this on my own machine; I can try a VM or something later.  Would you be able to try the tests with the following modification based on your suggestion?

```diff
diff --git a/t/11-replay-lwp.t b/t/11-replay-lwp.t
index 88c3410..19482a7 100644
--- a/t/11-replay-lwp.t
+++ b/t/11-replay-lwp.t
@@ -19,5 +19,5 @@ test_psgi $recorder_app, sub {
     $cb->(GET '/', Host => 'localhost');
 };
 
-my $output = qx(bin/plack-replay $requests_file t/reply-lwp.psgi -v);
+my $output = qx($^X bin/plack-replay $requests_file t/reply-lwp.psgi -v);
 like $output, qr/200 OK/, 'Using LWP::UserAgent to access an external service should succeed';
```

**Comment by eserte** [comment]
Tried it, with the patch there's no fail anymore.

**Comment by hoelzro** [comment]
Alright, I'll add that patch and make a trial release; that'll trigger a new build, right? Out of curiosity, what distro is the test machine running? 

On April 12, 2015 12:53:07 PM CDT, "Slaven Rezić" <notifications@github.com> wrote:
>Tried it, with the patch there's no fail anymore.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/hoelzro/plack-middleware-recorder/issues/9#issuecomment-92095719

-- 
Sent by Kaiten mail, typed with my left hand, and sent over sketchy WiFi. Please excuse my brevity.

**Comment by eserte** [comment]
Yes, most smokers test also trial/dev releases. My systems run freebsd 9, freebsd 10, debian wheezy and debian jessie, but there are a lot of other smokers around.

**Comment by hoelzro** [comment]
@eserte Alright, I've made the trial release!

**Comment by eserte** [comment]
http://matrix.cpantesters.org/?dist=Plack-Middleware-Recorder%200.05-TRIAL looks mostly OK. The only failing systems are Windows-like systems and where testers run the test suite as root, so the "chmod 0" trick in t/04-bad-filename-vcr.t does not work.

**Comment by hoelzro** [comment]
@eserte Thanks for checking and the Windows tip; do you any ideas on how I could get that test to pass on Windows?

**Comment by eserte** [comment]
Sorry, my knowledge of Windows is limited. Maybe it's possible to remove read permissions with the "ATTRIB" command.

**Comment by hoelzro** [comment]
@eserte I'll give that a shot when my dual boot machine is back up and running, then.
